Style/linkta 127 style landingpage and children components

dcpavel opened the pull request:
<!--# Pull Request Template -->

## Description

Adding styling to the landing page

<!--
Please include a summary of the change and which issue is fixed. Include relevant motivation and context. List any dependencies that are required for this change.

Linking Issues:
 - To associate this pull request with a GitHub issue within the same repository, use the "Fixes #IssueNumber" syntax.
 - For external issues, use Markdown links: [Issue Name](Issue Link).
-->



## Type of change

<!--Please delete options that are not relevant.-->

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [x] Other 🗒️ Styling

## How Has This Been Tested?

<!--Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Remove if not applicable. -->

I have visually tested this against the Figma designs:
https://www.figma.com/file/9QKsvDQ9jivgA9XfS0oqPd/Design-System---Linkta?type=design&node-id=1245-5890&mode=design&t=szrSlyw9xbsKWNIe-0

I used Chrome with a width of less than 390px. This was a width I thought I recalled as being the tipping point in our responsive design.

## Checklist:

<!--Before submitting your pull request, please review the following checklist:-->

- [x] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I have reviewed the [**Security Best Practices Document**](SECURITY_BEST_PRACTICES.md).
- [x] I have followed the [**Naming Conventions Guide**](NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

## Additional Information:

If someone wants to see what it looks like, give it a go. I am leaving this as a draft as I haven't integrated Jordan's recent PR: https://github.com/Linkta-org/landing-page/pull/22

This PR hasn't been approved. Hopefully we can get this done quickly. 

<!--Any additional information, configuration, or data that might be necessary to reproduce the issue or feature.-->

dcpavel committed:
🌈 style(Mobile Styling):

Begin styling by focusing on mobile layout

dcpavel committed:
Merge branch 'feature/LINKTA-140-implement-prelaunch-sign-up-form' of https://github.com/Linkta-org/landing-page into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
continuing to work on styling

dcpavel committed:
Continuing to style

dcpavel committed:
Merge branch 'feature/LINKTA-140-implement-prelaunch-sign-up-form' of https://github.com/Linkta-org/landing-page into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
Merge branch 'dev' of https://github.com/Linkta-org/landing-page into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
More styling changes

dcpavel committed:
Merge branch 'dev' into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
🌈 style(Styling for Mobile):

Complete styling for mobile, began styling for desktop. This may cause problems on larger screens.

dcpavel committed:
Putting back in the current flowState implementation.

dcpavel committed:
Merge branch 'feature/LINKTA-292-integrate-firebase-auth-with-prelaunch-signup-form' of https://github.com/Linkta-org/landing-page into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
Replace svg with transparent background in header, align Linkta to bottom of svg.

dcpavel committed:
Centering footer content, and spacing evenly.

dcpavel committed:
Center all content

dcpavel committed:
Fix dark mode coloring.

dcpavel committed:
Adding Privacy Statement to early-access and reducing text size.

dcpavel committed:
Fixing Typos

dcpavel committed:
more typographical fixes.

dcpavel committed:
make center text smaller

dcpavel committed:
More minor changes.

lomeintheory committed:
centered text; bolded error code; button border

lomeintheory committed:
centered text; bolded error code; button border

lomeintheory committed:
centered button

dcpavel committed:
Merge pull request #25 from Linkta-org/style/LINKTA-295-style-error-and-notfound-components

Style/LINKTA-295 style error and notfound components

dcpavel committed:
Merge branch 'dev' into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
Merge branch 'style/linkta-127-style-landingpage-and-children-components' of https://github.com/Linkta-org/landing-page into style/linkta-127-style-landingpage-and-children-components

ejwoods commented:
Looking into an issue with readability of text in input fields on dark mode

ejwoods committed:
add tailwind classes to style TextInput text with accent color in dark theme

This is a temporary fix to make text in the input fields readable in dark mode. Label text should retain light coloring, asterisk should ideally be red, and internal input field text should be differentiated from field background. Will revisit in a later commit.

ejwoods commented on app/layout.tsx:
```diff
@@ -19,10 +18,15 @@ export default function RootLayout({
-        <ColorSchemeScript />
       </head>
-      <body className={inter.className}>
-         <MantineProvider>{children}</MantineProvider>
```
Can you expand on the move of `<MantineProvider>` from `layout.tsx` to the root path's `page.tsx` and `PrelaunchSignUpForm.tsx`? 

From [Mantine docs](https://mantine.dev/theming/mantine-provider/):

> It must be rendered at the root of your application and should be used only once.

dcpavel replied:
Yes, when I used [createTheme](https://mantine.dev/theming/theme-object/ ) to handle styling the input elements, it did not like being at layout.tsx. I moved it over to PreluanchSignupForm.tsx originally to use the theme there. However, there is a button on the LandingPageTreeVisualizationPanel that is handled by mantine as well. I guess I should have moved the theme into page.tsx at the root, but I wasn't sure about passing the theme from the root for that single page's input fields.

dcpavel replied:
There is probably a better method to handle this, and I would love to know if you have some input.

ejwoods commented on app/layout.tsx:
```diff
@@ -19,10 +18,15 @@ export default function RootLayout({
   return (
     <html lang="en">
       <head>
-        <ColorSchemeScript />
```
[Matine docs](https://mantine.dev/theming/color-schemes/#colorschemescript) note:

> `ColorSchemeScript` . . . is used to avoid flash of inaccurate color scheme in server side rendered applications, for example Next.js or Remix.

Is `ColorSchemeScript` compatible with the theming you implemented?

dcpavel replied:
It was causing a bunch of styling overwriting and oddities. A lot of the documentation I read had talked about switching to HeadlessMantineProvider, but that created new issues as I was working. When I read through [the documentation here](https://v7.mantine.dev/theming/color-schemes) it didn't seem to be necessary to include it. If there's a better way of handling this, please let me know.

ejwoods reviewed (commented):
I'm getting 2 errors on the home page: 

```
Error: Text content does not match server-rendered HTML.

Warning: Text content did not match. Server: "body &gt; header { display: block; }" Client: "body > header { display: block; }"
```

```
Error: There was an error while hydrating this Suspense boundary. Switched to client rendering.
```

I deleted my .next, node_modules, and package-lock.json and re-ran `npm install` and `npm run dev` but still received these errors. I saw that some hydration issues were mentioned in slack, although I didn't see these errors mentioned specifically - maybe this is already being worked on?

dcpavel commented:
> I'm getting 2 errors on the home page:
> 
> ```
> Error: Text content does not match server-rendered HTML.
> 
> Warning: Text content did not match. Server: "body &gt; header { display: block; }" Client: "body > header { display: block; }"
> ```
> 
> ```
> Error: There was an error while hydrating this Suspense boundary. Switched to client rendering.
> ```
> 
> I deleted my .next, node_modules, and package-lock.json and re-ran `npm install` and `npm run dev` but still received these errors. I saw that some hydration issues were mentioned in slack, although I didn't see these errors mentioned specifically - maybe this is already being worked on?

This is what was being spoken about in #frontend earlier today. I think it's because I'm including a <script> in the HTML, but I'm also curious to track down what's breaking.

dcpavel committed:
Fixing hydration error

ejwoods committed:
pass handleSubmit to PrelaunchSignUpForm's onSubmit

This change was made in commit b103d48 but may have been mistakenly discarded during a merge.

dcpavel reviewed (commented):
good catch. (in response to Jordan's commit 86b7002)

ejwoods committed:
add ColorSchemeScript and MantineProvider to layout

ejwoods committed:
remove MantineProvider from root page.tsx

ejwoods committed:
remove MantineProvider from PrelaunchSignUpForm, convert theme to individual styles

Removed MantineProvider from form component after moving it layout. Applied theming previously implemented via createTheme with individual classNames props on target components.

ejwoods committed:
Merge branch 'dev' into style/jordan-test-linkta-127-branch

ejwoods committed:
Merge branch 'dev' into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
Merge pull request #28 from Linkta-org/style/jordan-test-linkta-127-branch

Updates to LINKTA-127 Style landing page and children components

dcpavel committed:
Merge branch 'dev' into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
Merge branch 'style/linkta-127-style-landingpage-and-children-components' of https://github.com/Linkta-org/landing-page into style/linkta-127-style-landingpage-and-children-components

dcpavel commented:
I just merged in the most recent changes to dev

lomeintheory reviewed (approved):
Styling looks good to me!

ejwoods commented on app/page.tsx:
```diff
@@ -1,19 +1,13 @@
-      </header>
-      <main>
+        <div className="sm:hidden ml-0">
+          <LinktaLogoWithText />
```
Kinski mentioned this in another PR - this div doesn't seem to be actually be rendering at any size. I'm not exactly sure why, but we already have the logo rendering in the `header` in `layout.tsx`. Can it be removed from this file or is there a need here?

ejwoods replied:
Tested removing this and don't see a change.

dcpavel replied:
I tested with this removed. There is no Linkta logo on the homepage when <= 390px width

ejwoods replied:
Ohhh I see. It looks like that behavior is coming from `globals.css`:

```
  @media screen and (max-width: 390px) {
    body > footer, body > header {
      @apply hidden;
    }
  ...other styling
  }
```

Removing `body > header` from this rule will fix that issue as well. Of course, it would also display anything else in our header at that screen size. Currently the logo is the only child element of the header. What do you think?

ejwoods replied:
Alternatively, if we want to keep this div to show the logo on the home page only (and not the sign up page) on mobile, we'll want to update the breakpoint. Because `sm` is used here, the element is hidden at 390px (iphone 12 width) but visible below.

ejwoods committed:
update dark mode styling of TextInput components to only apply above screen size "sm"

ejwoods committed:
set custom screen size breakpoint for input text

This is a quick workaround that needs revisiting. Dark mode styling is currently not being applied correctly to screens 390px and below, but input text styling is applied inline. This caused an issue where dark mode styling for input text was being applied without the rest of dark mode styling. Setting a custom breakpoint of 391px for the input text fixes that issue for now.

ejwoods commented on app/globals.css:
```diff
@@ -1,5 +1,74 @@
+
+  @media screen and (max-width: 390px) {
+    body {
+      @apply bg-[#F4FBF9] m-auto text-black min-w-[200px];
```
I'm just catching onto the fact that we are specifically overriding dark mode for "mobile" users 😅 

We might consider increasing the `max-width` for this media query to `430px` to be consistent with larger phones.

dcpavel committed:
Merge branch 'dev' into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
Merge branch 'style/linkta-127-style-landingpage-and-children-components' of https://github.com/Linkta-org/landing-page into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
Reworking styling to conform to tailwind styles.

dcpavel committed:
Merge branch 'feature/LINKTA-305-Update-PrelaunchSignUpForm-to-include-additional-input-fields' of https://github.com/Linkta-org/landing-page into feature/LINKTA-305-Update-PrelaunchSignUpForm-to-include-additional-input-fields

dcpavel committed:
Merge branch 'feature/LINKTA-305-Update-PrelaunchSignUpForm-to-include-additional-input-fields' into style/linkta-127-style-landingpage-and-children-components

dcpavel committed:
🌈 style: Landing Page

Completing style conversion to standard tailwind styling. Reduced rules in globals.css, moving most rules to classes on elements.

Some prettier formating changes were made as well.

dcpavel committed:
Minor PrelaunchSignUpForm modifications to match Figma

ejwoods committed:
Merge branch 'dev' into style/linkta-127-style-landingpage-and-children-components

ejwoods committed:
add PrivacyAgreement back to PrelaunchSignUpForm

ejwoods commented on tailwind.config.ts:
```diff
@@ -6,7 +6,20 @@ const config: Config = {
   theme: {
+    screens: {
+      // defaults from https://tailwindcss.com/docs/theme
+      sm: '431px', // 430px is the width of the iPhone 14 Pro Max
```
Know I commented about using this size for our mobile breakpoint before - I actually believe tailwind's default `sm` breakpoint of `640px` encourages mobile styling to extend all the way to that screen width, where they expect tablet sizing to begin.

Generally I think it's best to use the default screen size breakpoints so that user experience of our site is consistent with other tailwind sites. 

I recommend removing this custom breakpoint unless we have a good reason to have non-mobile styling kick in at exactly 431px.

dcpavel replied:
I can get behind this. @ChTiSh, @kinskiwu What do you think about this suggestion.

ejwoods commented on tailwind.config.ts:
```diff
@@ -6,7 +6,20 @@ const config: Config = {
+      // We currently have web and mobile breakpoints
+      // md: '768px',
+      // lg: '976px',
+      // xl: '1440px',
```
These are not in use currently, but like above, I recommend removing custom breakpoints and using tailwind's defaults.

ejwoods commented on app/components/main-content/LandingPageMainHero.tsx:
```diff
@@ -11,9 +11,17 @@ export default function LandingPageMainHero() {
+          onClick={handleClick}
+          label="Join Our Journey"
+          classNames={{
+            root: 'button-primary sm:flex-o sm:align-self-end',
```
Having trouble figuring out what `flex-o` should be doing. I believe `align-self-end` should be `self-end` in tailwind.

ejwoods committed:
remove dark theme TextInput text color styling for sm screens

Removed styling changing the text color of TextInput elements at 'sm' screen size breakpoint. This was an old workaround that is no longer needed.

ejwoods reviewed (approved):
I looked into removing the custom screen breakpoints from the tailwind config and tailwind's defaults don't seem to be applying correctly. For example, `sm` styling does not apply at the default of `640px` unless explicitly set. This is something to revisit in a future PR. 

I think we are good to merge this for now so a new styling PR can be opened.

ejwoods committed:
Merge branch 'dev'

Following PR 38, removed useEffect and OAuth sign in section from form, removed h3 element from email sign up section in form.

ejwoods committed:
add dark theme text styling for input fields in globals.css

Input fields in dark mode previously inherited light text coloring from their parent components. This made input text difficult to read against the white background of the input elements. Set the text color to black in globals.css styling for input components to enhance readability.
